File: run_tnbc.R
source("process_data.R")
source("methods.R")

args <- commandArgs(trailingOnly = TRUE)
vis_id <- args[[1]] # e.g., "1142243F"
method <- args[[2]] # e.g., DOT, CARD, RCTD

sc_data <- load_sc_wu('TNBC')
vis_data <- load_vis_tnbc(vis_id)
annotations <- sc_data$metaa[, "celltype_major"]
locations <- vis_data$meta[, c("Col", "Row")]

if(method == "DOT")
{
  weights <- run_DOT(sc_data$counts, annotations, vis_data$counts, locations, normalize = FALSE)
}else if(method == "CARD")
{
  weights <- run_CARD(sc_data$counts, annotations, vis_data$counts, locations)  
}else if(method == "RCTD")
{
  weights <- run_RCTD(sc_data$counts, annotations, vis_data$counts, locations)
}

File: run_tnbc.py
from process_data import load_sc_wu, load_vis_tnbc
import sys

args = sys.argv
vis_id = args[1] # e.g., "1142243F"
method = args[2] # e.g., TACCO, cell2location, ...

ref_adata = load_sc_wu("TNBC")
ref_annotation = "celltype_major"
srt_adata = load_vis_tnbc(vis_id)

if method.lower() == "tacco":
    from methods import run_tacco
    weights = run_tacco(ref_adata, srt_adata, ref_annotation)
elif method.lower() in ["cell2location", "c2l"]:
    from methods import run_cell2location
    weights = run_cell2location(ref_adata, srt_adata, ref_annotation, 
                                epochs = 20000, normalize = False)